### DISCUSSION TITLE
Assign role after login

### DISCUSSION THREAD

**Discussion by fornaserofranco** [spatie/laravel-permission#2150] [Q&A]
I am implementing roles in a website where there is already a database with many users, so assigning roles one by one is not feasible. It occurred to me to assign a role to the user after they log in, but I can't get it to work. Could you tell me what is wrong with my code?

This is my login controller

```php
if (Auth::attempt(['email' => $request->email, 'password' => $request->password,'codigocli'=>$request->codigocli])) 
        {   
            $user = User::where('id', '=', Auth::user()->id)
                        ->first();

            switch($user->codigocli){
                case 0:
                    $user->assignRole('vendedor');
                    break;
                case 7777:
                    $user->assignRole('admin');
                    break;
                default:
                    $user->assignRole('cliente');
                    break;
            }

            return redirect()->route('index');
        }

```

The error I get is:
![image](https://user-images.githubusercontent.com/108434960/182860290-c031e1f0-bfd3-4cfb-82b6-de3665ca27e1.png)

I already have HasRole in my model user and was able to assign a role to a user in a seeder the way I'm trying to do now

**Comment by erikn69** [comment]
Really weird
https://spatie.be/docs/laravel-permission/v5/basic-usage/basic-usage
Take a look on `HasRole` again
